File: Data Structures & Algorithms/palindromic-substrings/submission-2.cpp
class Solution {
public:
    int countSubstrings(string s) {
        vector<vector<bool>> pallindromeGrid(s.size(), vector<bool>(s.size(), false));
        int count = 0;
        for (int i = pallindromeGrid.size() - 1; i >= 0; i--) {
            for (int j = i; j < pallindromeGrid[0].size(); j++) {
                if (i == j) {
                    pallindromeGrid[i][j] = true;
                    count += 1;
                } else if (j - i == 1) {
                    if (s[i] == s[j]) {
                        pallindromeGrid[i][j] = true;;
                        count += 1;
                    }
                } else {
                    if ((s[i] == s[j]) && (pallindromeGrid[i + 1][j - 1])) {
                        pallindromeGrid[i][j] = true;
                        count += 1;
                    }
                }
            }
        }
        return count;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-2.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagramSet = {}
        for word in strs:
            #create alphabet count for this word
            alphabetCount = [0] * 26
            for letter in word:
                alphabetCount[ord(letter)-ord('a')] += 1
            #check if alphabet count already exists
            if tuple(alphabetCount) in anagramSet:
                anagramSet[tuple(alphabetCount)].append(word)
            else:
                anagramSet[tuple(alphabetCount)] = [word]
        
        #print result
        result = []
        for key, value in anagramSet.items():
            result.append(value)
        #return result
        return result